temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/wallet/AmountCurrency.kt

```diff
@@ -66,21 +73,25 @@ fun AmountCurrencyB1Row(
     }
 }
 
-@SuppressLint("ComposeContentEmitterReturningValues")
+@SuppressLint("ComposeContentEmitterReturningValues", "CoroutineCreationDuringComposition")
 @Composable
 fun AmountCurrencyB1(
     amount: Double,
     currency: String,
     amountFontWeight: FontWeight = FontWeight.Bold,
     textColor: Color = UI.colors.pureInverse,
-    shortenBigNumbers: Boolean = false,
     hideIncome: Boolean = false
 ) {
-    val shortAmount = shortenBigNumbers && shouldShortAmount(amount)
+    val context = LocalContext.current
+    val scope = rememberCoroutineScope()
+    var formattedAmount by remember { mutableStateOf(amount.toDecimalFormatWithFraction()) }
+    scope.launch {
```

Comment:
Use LaunchedEffect this will result in countless coritines launched when recomposition happens

Reply:
Done